collector/mssql.go

```diff
@@ -246,1021 +244,1026 @@ type MSSQLCollector struct {
+			"(BufferManager.Buffercachehitratio)",
 			[]string{"instance"},
 			nil,
 		),
 		CheckpointpagesPersec: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "checkpointpages_persec"),
-			"(CheckpointpagesPersec)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_checkpoint_pages"),
+			"(BufferManager.CheckpointpagesPersec)",
 			[]string{"instance"},
 			nil,
 		),
 		Databasepages: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "databasepages"),
-			"(Databasepages)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_database_pages"),
+			"(BufferManager.Databasepages)",
 			[]string{"instance"},
 			nil,
 		),
 		Extensionallocatedpages: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "extensionallocatedpages"),
-			"(Extensionallocatedpages)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_extension_allocated_pages"),
+			"(BufferManager.Extensionallocatedpages)",
 			[]string{"instance"},
 			nil,
 		),
 		Extensionfreepages: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "extensionfreepages"),
-			"(Extensionfreepages)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_extension_free_pages"),
+			"(BufferManager.Extensionfreepages)",
 			[]string{"instance"},
 			nil,
 		),
 		Extensioninuseaspercentage: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "extensioninuseaspercentage"),
-			"(Extensioninuseaspercentage)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_extension_in_use_as_percentage"),
+			"(BufferManager.Extensioninuseaspercentage)",
 			[]string{"instance"},
 			nil,
 		),
 		ExtensionoutstandingIOcounter: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "extensionoutstanding_i_ocounter"),
-			"(ExtensionoutstandingIOcounter)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_extension_outstanding_io_counter"),
+			"(BufferManager.ExtensionoutstandingIOcounter)",
 			[]string{"instance"},
 			nil,
 		),
 		ExtensionpageevictionsPersec: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "extensionpageevictions_persec"),
-			"(ExtensionpageevictionsPersec)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_extension_page_evictions"),
+			"(BufferManager.ExtensionpageevictionsPersec)",
 			[]string{"instance"},
 			nil,
 		),
 		ExtensionpagereadsPersec: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "extensionpagereads_persec"),
-			"(ExtensionpagereadsPersec)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_extension_page_reads"),
+			"(BufferManager.ExtensionpagereadsPersec)",
 			[]string{"instance"},
 			nil,
 		),
 		Extensionpageunreferencedtime: prometheus.NewDesc(
-			prometheus.BuildFQName(Namespace, subsystem, "extensionpageunreferencedtime"),
-			"(Extensionpageunreferencedtime)",
+			prometheus.BuildFQName(Namespace, subsystem, "bufman_extension_page_unreferenced_time"),
```

Comment:
Do you know what this measures? Is it absolute time or relative?

Reply:
According to [Microsoft](https://docs.microsoft.com/en-us/sql/relational-databases/performance-monitor/sql-server-buffer-manager-object?view=sql-server-2017):
`Average seconds a page will stay in the buffer pool extension without references to it.`

So I think a Gauge type is appropriate. 

Changed name to `bufman_extension_page_unreferenced_seconds`